src/app/installer/create-wallet/create-wallet.component.ts

```diff
@@ -83,8 +95,13 @@ export class CreateWalletComponent implements OnDestroy {
   }
 
   initialize(type: number): void {
+    const wName: string = this.name;
```

Comment:
combined with this?

Reply:
I see now, I don't know how I skipped over that line.
Perhaps this.reset() can be called in ngOnInit, such that it doesn't overwrite it at all.
AFAIK it only sets it to the default values.

Reply:
`this.reset()` is called in the constructor, which could be moved to ngOnInit as well. `initialize()` however, is also used as the click handler for both the 'restore' and 'create' buttons, and so we likely want to be calling `this.reset()` whenstarting either the create or restore wallet process. Particularly if the user has already reached the first step for a new wallet, prinput a name and hit the 'back' button. In which case we need to be cleaning up things like error messages, the wallet name, etc. So `this.reset()` should probably be used here as well. Which then necessitates the need to keep the wallet name (only) if the wallet is an existing wallet.